src/components/source_editor/LatticeConfigurationDialog.js

```diff
@@ -108,6 +111,22 @@ class LatticeConfigurationDialog extends React.Component {
         const newMaterial = new Material(newMaterialConfig);
         // assert basis is stored in 'crystal' units
         newMaterial.toCrystal();
+
+        // update to non-periodic if asked to do so
+        if (this.state.isNonPeriodic) {
+            const newBasis = new Made.Basis({
+                elements: this.props.material.basis.elements,
+                coordinates: this.props.material.basis.coordinates,
+            });
+            let maxPairwiseDistance = newBasis.maxPairwiseDistance;
```

Comment:
Made.Basis is where we calculate the max pairwise distance for a molecule, which is why we need a Basis object here.